Compilation Error with Rust 1.39.0

Trying to upgrade my project to use the latest release this morning the MSRV job is failing on compiling the numpy 0.12.0 crate:

https://github.com/Qiskit/retworkx/pull/139/checks?check_run_id=1238063212#step:6:257

```
error[E0210]: type parameter `T` must be used as the type parameter for some local type (e.g., `MyStruct<T>`)
      --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/numpy-0.12.0/src/array.rs:113:1
      |
  113 | pyobject_native_type_named!(PyArray<T, D>, T, D);
      | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ type parameter `T` must be used as the type parameter for some local type
      |
      = note: only traits defined in the current crate can be implemented for a type parameter
      = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
  
  error: aborting due to previous error
```

The CI other jobs using Rust 1.47.0 compile fine.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 # Changelog
 
+- v0.12.1
+  - Fix compile error in Rust 1.39
+
 - v0.12.0
   - Introduce `NpySingleIter` and `NpyMultiIter`.
   - Introduce `PyArrayDescr`.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "numpy"
-version = "0.12.0"
+version = "0.12.1"
 authors = ["Toshiki Teramura <toshiki.teramura@gmail.com>", "Yuji Kanagawa <yuji.kngw.80s.revive@gmail.com>"]
 description = "Rust binding of NumPy C-API"
 documentation = "https://pyo3.github.io/rust-numpy/numpy"

--- a/README.md
+++ b/README.md
@@ -58,7 +58,7 @@ name = "numpy-test"
 
 [dependencies]
 pyo3 = "0.12"
-numpy = "0.12.0"
+numpy = "0.12"
 ```
 
 ```rust
@@ -92,7 +92,7 @@ name = "rust_ext"
 crate-type = ["cdylib"]
 
 [dependencies]
-numpy = "0.12.0"
+numpy = "0.12"
 ndarray = "0.13"
 
 [dependencies.pyo3]

